### DISCUSSION TITLE
Cleaification /api-routes, .data.ts vs routeData()

### DISCUSSION THREAD

**Discussion by benbender** [solidjs/solid-start#194] [Q&A] [answered]
Basically I'm confused which concepts for data-fetching are the way forward, which are - if any - deprecated and - if any - what are their pros and cons?

I totally understand that this project is still in its early days, but a hint/doc/discussion/example would be greatly appreciated!

**Comment by ryansolid** [comment, marked as answer]
`.data.ts` vs routeData are the same thing. Just one is in a separate file and the other is inlined. I imagine final docs won't even mention `.data.ts` as it is more of a mechanical thing. Might consider removing it.

API routes are a bit different. They are designed for creating APIs, mostly for external consumption. For the most part you can just use routeData in SolidStart and never write an API. But if you need an API for another service this is a way you can do it.

Docs are coming. They are actually more of the bottleneck for the release than anything. They take a lot of time (more than the development) and as soon as they are ready we will announce beta release.
